src/compiler/checker.ts

```diff
@@ -2836,11 +2836,16 @@ namespace ts {
         // pattern. Otherwise, it is the type any.
         function getTypeFromBindingElement(element: BindingElement, includePatternInType?: boolean): Type {
             if (element.initializer) {
-                return getWidenedType(checkExpressionCached(element.initializer));
+                const type = checkExpressionCached(element.initializer);
+                reportErrorsFromWidening(element, type);

```

Comment:
is this needed to make the new tests fail properly? If not, then does there need to be some test case that does need it to get the correct error message?